File: TechRadar.Web/ClientApp/app/models/blip.ts
import moment = require("moment");

export class Blip {
    id: string;
    name: string;
    description: string;
    added: Date;
    size: number;
    quadrantId: string;
    cycleId: string;
    radarId: string;
    isNew: boolean;
    blipNumber: number;

    constructor(id: string, name: string, description: string, size: number, added: Date, cycleId: string, quadrantId: string, radarId: string) {
        this.id = id;

        this.name = name;
        this.description = description ? description : "";
        this.added = added;
        this.size = size ? size : 1;

        this.cycleId = cycleId;
        this.quadrantId = quadrantId;

        this.radarId = radarId;

        var diff = moment(new Date()).diff(moment(this.added), "months");
        this.isNew = diff <= 3;

        this.blipNumber = 0;
    }
}

File: TechRadar.Services/Models/Blip.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace TechRadar.Services.Models
{
    [BsonIgnoreExtraElements]
    public class Blip
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [BsonElement("name")]
        public string Name { get; set; }

        [BsonElement("description")]
        public string Description { get; set; }

        [BsonElement("added")]
        [BsonDateTimeOptions(DateOnly =true)]
        public DateTime Added { get; set; }

        [BsonElement("size")]
        public int Size { get; set; }

        [BsonElement("radar")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string RadarId { get; set; }

        [BsonElement("quadrant")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string QuadrantId { get; set; }

        [BsonElement("cycle")]
        [BsonRepresentation(BsonType.ObjectId)]
        public string CycleId { get; set; }
    }
}
